urlextract/urlextract_core.py

```diff
@@ -745,13 +717,18 @@ def gen_urls(self, text, check_dns=False, get_indices=False):
             # move cursor right after found TLD
             tld_pos += len(tld) + offset
 
-    def find_urls(self, text, only_unique=False, check_dns=False, get_indices=False):
+    def find_urls(self, text, only_unique=False, check_dns=False, get_indices=False, timeout=None,
+                  accept_on_timeout=False, max_workers=None, max_tasks=None):
```

Comment:
What I am thinking here is, isn't there already too much parameters for this find function?
Do you think that it is not sufficient to let user set these parameters ahead by modifying properties of `DNSCheck` class

What is you opinion?

Reply:
We could write some setter methods for the dsn_checker class in the URLExtract class and the user could set then before running `find_urls`. But I would add to the docs of `find_urls` the default values for dns_checker in case the user sets `check_dns = True` and is not aware of it. What do you think?

Reply:
I vote for setter methods in dns_check class. Ideally all methods related to DNS checks should be in that class. (not directly in URLExtract class) 
And yes, everything should be documented. And also everything should have reasonable default values.